lib/gpu_memory_service/integrations/vllm/worker.py

```
@@ -216,13 +240,24 @@ def sleep(self, level: int = 1) -> None:
         2. Shadow: KV cache was skipped at startup, manager has no allocations
            (unmap_all_vas is a no-op, abort disconnects)
         """
         free_bytes_before = torch.cuda.mem_get_info()[0]
 
+        # Unpublish MX metadata before GMS unmap (CUDA handles still valid).
+        # Stops heartbeat + worker gRPC server, marks STALE on MX server.
+        mx_ctx = get_mx_load_context()
+        if mx_ctx is not None:
+            try:
+                unpublish_metadata(mx_ctx)
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)